hazelcast/src/test/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManagerTest.java

```diff
@@ -80,4 +82,30 @@ private void verifyIsUnisocketClient(boolean tpcEnabled, boolean smartRouting) {
         // should be unisocket only when smart routing is false and TPC disabled
         assertEquals(!smartRouting && !tpcEnabled, isUnisocket);
     }
+
+    @Test
+    public void testSkipMemberListDuringReconnection() {
+        HazelcastInstance instance = factory.newHazelcastInstance(smallInstanceConfigWithoutJetAndMetrics());
+
+        Address address = instance.getCluster().getLocalMember().getAddress();
+        String addressString = address.getHost() + ":" + address.getPort();
+        ClientConfig config = new ClientConfig();
+        config.setProperty(TcpClientConnectionManager.SKIP_MEMBER_LIST_DURING_RECONNECTION.getName(), "true");
+        config.getNetworkConfig().setSmartRouting(false);
+        config.getNetworkConfig().addAddress(addressString);
+        config.getConnectionStrategyConfig().getConnectionRetryConfig().setClusterConnectTimeoutMillis(3_000);
+
+        // There are two members, and the unisocket client is connecting
+        // to one of them. (the address of the `instance` defined above)
+        HazelcastInstance client = factory.newHazelcastClient(config);
+
+        assertEquals(2, client.getCluster().getMembers().size());
+        instance.shutdown();
+
+        // We shut down the `instance` the client is connected to but
+        // there is still a member running. If the client was to try to
+        // connect to members from the member list, it would succeed
+        // and the assertion below would never be true.
+        assertTrueEventually(() -> assertFalse(client.getLifecycleService().isRunning()));
```

Comment:
```suggestion
        assertFalseEventually(client.getLifecycleService().isRunning());
```

Reply:
It doesn't work the way you wrote. `assertFalseEventually` returns without an error when the assertion fails eventually. So, I would have to write it as  `assertFalseEventually(() -> assertTrue(client.getLifecycleService().isRunning()));` which is not much different than the current version. So, if it is OK for you, I will stick with the current version, as it is widely used in the code base